test/index.test.js

```diff
@@ -187,18 +220,12 @@ describe("iw-mysql-wrapper", function () {
 						transactionScope.query("INSERT INTO `iw_mysql_wrapper_test` VALUES (1);"),
 						transactionScope.query("BAD SQL;")
 					]);
-				}).then(function () {
-
-					var err = new Error("Transaction should have failed!");
-					err.code = "E_DID_NOT_FAIL";
-					throw err;
-
-				}).catch(function verifyAfterRollback(err) {
+				}).then(neverCallMe("Query should throw a parse error.")).catch(function verifyAfterRollback(err) {
 
 					expect(err.code).to.eql("ER_PARSE_ERROR");
 					return Q.all([
-						savedScope.query("SELECT * FROM `iw_mysql_wrapper_test`").then(function (rows) {
-							expect(rows).to.eql([], "Transaction scope: should be rolled back");
+						savedScope.query("SELECT 1;").then(neverCallMe("Transaction should be closed.")).catch(function (err) {
+							expect(err.code).to.eql("E_TRANSACTION_CLOSED");
```

Comment:
There are several asserts in here:
1. That parse error was rethrown
2. That you can no longer execute queries inside the transaction
3. That the transaction was not committed whatsoever - i.e. the table is empty.

If you have a better idea how to assert that the transaction was rolled back - let me know.